### PR TITLE
test: add an integration test for new row format

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
add an integration test for new row format

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-cdc/cdc-ansible`
 - Need to be included in the release note

### CODE REVIEW

File: tests/row_format/data/prepare.sql [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+VALUES ( false, 666, 123.777, 123456789012.123456789012, b'1000001'
+       , '1000-01-01', '9999-12-31 23:59:59', '19731230153000', '23:59:59', 1970
+       , '测', '测试', 'blob', '测试text11', 'enum3'
+       , 'a,b', NULL);
+
+UPDATE multi_data_type
+SET t_bigint = 555
+WHERE id = 1;
+
+SET GLOBAL tidb_row_format_version = 2;
```
Comment (amyangfei) [inline tests/row_format/data/prepare.sql]: what is the purpose of inserting data with `new format` -> `old format` -> `new format`
Is it used to simulate different TiDB write with either new or old format?

Comment (leoppro) [reply tests/row_format/data/prepare.sql]: `old format` -> `new format`: simulate the tidb cluster of upgrades from 3.x to 4.x;
`new format` -> `old format` and `old format` -> `new format`: simulate the user adjusts tidb_row_format_version parameter manually

**Pull request merged**